client/core/core.go

```diff
@@ -2614,12 +2607,27 @@ func (c *Core) registerHD(dc *dexConnection, crypter encrypt.Crypter) (*incomple
 	}
 
 	creds := c.creds()
-	encKey, acctPubB, err := dc.acct.setupCryptoV2(creds, crypter)
-	if err != nil {
-		return nil, false, newError(acctKeyErr, "setupCryptoV2 error: %v", err)
-	}
+	for {
+		encKey, acctPubB, err := dc.acct.setupCryptoV2(creds, crypter)
+		if err != nil {
+			return nil, false, newError(acctKeyErr, "setupCryptoV2 error: %v", err)
+		}
+
+		incompleteReg, paid, suspended, err := c.register(dc, encKey, nil, acctPubB)
+		if err == nil && suspended {
+			// Reset the account keys and try again. A new key pair will be
+			// derived. If the available key pool for this DEX is exhausted,
+			// dc.acct.setupCryptoV2 will produce an error.
+			// TODO: If the user did not attempt account discovery, this
+			// suspended account may have active matches that the user will
+			// not be aware of.
+			c.log.Infof("HD account key for %s was reported as suspended. Deriving another key to use.", dc.acct.host)
+			dc.acct.resetKeys()
+			continue
```

Comment:
Let's add a something like 200 ms sleep here so we don't get our requests denied on account of request rate.